### PR TITLE
Batch utilities

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

Ports Batch Utilities for SeqToSeq.
Tasks:
- [x] batch_elements
- [x] input_array
- [x] output_array
- [x] tests
- [x] add to docs

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/utils/batch_utils.py [hunk 1]
```diff
@@ -129,3 +130,159 @@ def batch_coulomb_matrix_features(X_b: np.ndarray,
...
+    Parameters
+    ----------
+    elements: Any
+        Elements to be combined into batches.
+    batch_size: int
+        Batch size in which to divide.
+
+    Returns
+    -------
+    batch: Any
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:147]: Each batch will be a List[Any]

File: deepchem/utils/batch_utils.py [hunk 2]
```diff
@@ -129,3 +130,159 @@ def batch_coulomb_matrix_features(X_b: np.ndarray,
...
+            batch = []
+    if len(batch) > 0:
+        yield batch
+
+
+def create_input_array(sequences: Collection, reverse_input: bool,
+                       batch_size: int, input_dict: Dict, end_mark: Any):
+    """Create the array describing the input sequences for a batch.
+
+    It creates a 2d Matrix empty matrix according to batch size and max_length.
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:175]: Why do we want to do this? Can you provide more context on the usage in seq2seq or other models?

File: deepchem/utils/batch_utils.py [hunk 3]
```diff
@@ -129,3 +130,159 @@ def batch_coulomb_matrix_features(X_b: np.ndarray,
...
+            features[i, j] = input_dict[token]
+    features[np.arange(len(sequences)), lengths] = input_dict[end_mark]
+    return features
+
+
+def create_output_array(sequences: Collection, max_output_length: int,
+                        batch_size: int, output_dict: Dict, end_mark: Any):
+    """Create the array describing the target sequences for a batch.
+
+    Create the array describing the output sequences for a batch.
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:238]: Same comment here about docstring

File: deepchem/utils/batch_utils.py (hunk 1 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:147]: Done

File: deepchem/utils/batch_utils.py (hunk 3 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py]: Done

File: deepchem/utils/batch_utils.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:175]: Done

File: docs/source/api_reference/utils.rst [hunk 4]
```diff
@@ -328,4 +334,4 @@ for additional information regarding equivariance and Deepchem's support for equ
 
 .. autofunction:: deepchem.utils.equivariance_utils.change_basis_real_to_complex
 
-.. autofunction:: deepchem.utils.equivariance_utils.wigner_D
\ No newline at end of file
+.. autofunction:: deepchem.utils.equivariance_utils.wigner_D
```
Comment (rbharath) [inline docs/source/api_reference/utils.rst:337]: This looks like cruft; let's rebase instead of merge

**Pull request closed without merging**